### PR TITLE
[FLINK-9904]Allow users to control MaxDirectMemorySize

### PR DESCRIPTION
**Pull request by tianchen92**
## What is the purpose of the change

*This PR is related to [FLINK-9904](https://issues.apache.org/jira/browse/FLINK-9904). Make MaxDirectMemorySize configurable via flink-conf.yaml.*

## Brief change log

  - *Make MaxDirectMemorySize configurable.*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (don't know)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by tianchen92** [comment]
@klion26 Thanks for your comments, I've created a new PR.

File: flink-dist/src/main/flink-bin/bin/config.sh [hunk 1]
```diff
@@ -429,6 +430,11 @@ if [ -z "${FLINK_TM_NET_BUF_MAX}" -o "${FLINK_TM_NET_BUF_MAX}" = "-1" ]; then
     FLINK_TM_NET_BUF_MAX=$(parseBytes ${FLINK_TM_NET_BUF_MAX})
 fi
 
+# Define FLINK_TM_MAX_OFFHEAP_SIZE if it is not already set
+if [ -z "${FLINK_TM_MAX_OFFHEAP_SIZE}" ]; then
+    # default: Long.MAX_VALUE in TB
+    FLINK_TM_MAX_OFFHEAP_SIZE=$(readFromConfig ${KEY_TASKM_MAX_OFFHEAP_SIZE} "8388607T" "${YAML_CONF}")
```
Comment (chunweilei) [inline flink-dist/src/main/flink-bin/bin/config.sh]: Should the condition be `if [ -z "${FLINK_TM_MAX_OFFHEAP_SIZE}" -o "${FLINK_TM_MAX_OFFHEAP_SIZE}" = "-1" ]`?

Comment (tianchen92) [reply flink-dist/src/main/flink-bin/bin/config.sh]: I am not sure. In what case would ${FLINK_TM_MAX_OFFHEAP_SIZE} be equal to -1?

Comment (chunweilei) [reply flink-dist/src/main/flink-bin/bin/config.sh]: Maybe it is set to -1 by user incorrectly. You can find other configurations like `FLINK_TM_NET_BUF_MAX` use this condition too.

Comment (tianchen92) [reply flink-dist/src/main/flink-bin/bin/config.sh]: Resolved, thanks!

**Comment by tianchen92** [comment]
Hi @zentol, could you please help review this PR? thanks very much

**Comment by zentol** [comment]
Subsumed by the new memory configuration options.

**Pull request closed without merging**